l1-contracts/test/foundry/integration/BaseIntegrationTests.t.sol

```diff
@@ -0,0 +1,263 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import {L2TransactionRequestDirect, L2TransactionRequestTwoBridgesOuter} from "contracts/bridgehub/IBridgehub.sol";
+import {TestnetERC20Token} from "contracts/dev-contracts/TestnetERC20Token.sol";
+
+import {MailboxFacet} from "contracts/state-transition/chain-deps/facets/Mailbox.sol";
+
+import {L1ContractDeployer} from "./_SharedL1ContractDeployer.t.sol";
+import {TokenDeployer} from "./_SharedTokenDeployer.t.sol";
+import {HyperchainDeployer} from "./_SharedHyperchainDeployer.t.sol";
+import {L2TxMocker} from "./_SharedL2TxMocker.t.sol";
+
+import {ETH_TOKEN_ADDRESS} from "contracts/common/Config.sol";
+
+contract BaseIntegrationTests is L1ContractDeployer, HyperchainDeployer, TokenDeployer, L2TxMocker {
+    address alice;
+    address bob;
+
+    TestnetERC20Token baseToken;
+
+    function setUp() public {
+        deployL1Contracts();
+        deployTokens();
+
+        registerNewTokens(tokens);
+
+        addNewHyperchainToDeploy("hyperchain1", ETH_TOKEN_ADDRESS);
+        addNewHyperchainToDeploy("hyperchain2", ETH_TOKEN_ADDRESS);
+        addNewHyperchainToDeploy("hyperchain3", tokens[0]);
+        addNewHyperchainToDeploy("hyperchain4", tokens[0]);
+        deployHyperchains();
+
+        alice = makeAddr("alice");
+        bob = makeAddr("bob");
+        baseToken = TestnetERC20Token(tokens[0]);
+    }
+
+    function test_hyperchainTokenDirectDeposit_Eth() public {
+        vm.txGasPrice(0.05 ether);
+        vm.deal(alice, 1 ether);
+        vm.deal(bob, 1 ether);
+
+        uint256 firstChainId = hyperchainIds[0];
+        uint256 secondChainId = hyperchainIds[1];
+
+        assertTrue(getHyperchainBaseToken(firstChainId) == ETH_TOKEN_ADDRESS);
+        assertTrue(getHyperchainBaseToken(secondChainId) == ETH_TOKEN_ADDRESS);
+
+        L2TransactionRequestDirect memory aliceRequest = createMockL2TransactionRequestDirect(
+            firstChainId,
+            1 ether,
+            0.1 ether
+        );
+        L2TransactionRequestDirect memory bobRequest = createMockL2TransactionRequestDirect(
+            secondChainId,
+            1 ether,
+            0.1 ether
+        );
+
+        bytes32 canonicalHash = keccak256(abi.encode("CANONICAL_TX_HASH"));
+        address firstHyperChainAddress = getHyperchainAddress(firstChainId);
+        address secondHyperChainAddress = getHyperchainAddress(secondChainId);
+
+        vm.mockCall(
+            firstHyperChainAddress,
+            abi.encodeWithSelector(MailboxFacet.bridgehubRequestL2Transaction.selector),
+            abi.encode(canonicalHash)
+        );
+
+        vm.mockCall(
```

Comment:
It would be better to avoid `mockCall` in the integration test. If you want to test e2e the `bridgehubRequestL2Transaction` function, you should just get a transaction hash after call. The transaction itself should be in format that `MailboxFacet:bridgehubRequestL2Transaction` expect to receive it.

Reply:
Understood, it was fixed in latest changes